### ISSUE TITLE
Deprecate support for python3.5

### ISSUE THREAD

**Issue by zagura** [cloudflare/python-cloudflare#190]
Using f-string and configuring support for python3.5 result in getting incompatible version via pip. When f-string were used, python3.5 won't parse the script.

**Comment by mahtin** [comment]
Do you need `Python3.5` support? It was unclear from your issue text. However, that said, yes ... there's an issue!

There's only one file with f-string being used and that's `examples/example_page_rules.py`. 
```python
    url_match = f"*.{zone_name}/url1*"
    url_forwarded = f"http://{zone_name}/url2"
```
That can easily be updated as the same operations are done without f-string's in other examples.

You are correct that f-string was [added](https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498) in `3.6` and hence `3.5` will barf at that file. According to [3.5.10 release](https://www.python.org/downloads/release/python-3510) "Python 3.5 has reached end-of-life. Python 3.5.10 is the final release of 3.5." so I wonder if dropping support for `3.5` wouldn't be a bad idea anyway. In fact, as this Cloudflare project is heavily dependent on [requests](https://pypi.org/project/requests/) and that's `3.7` and above, I think it's worthy to update Cloudflare's setup to be `3.7` and above. 

Thoughts? (as I don't want to action anything until I understand this full issue).

**Comment by zagura** [comment]
Thanks for quick response. 
When installing package with `pip install cloudflare` it would select the newest version which matches criteria (including interpreter version. Keeping 3.5.0 is pointless as it doesn't work with python3.5, therefore pip choose "broken" version. 
In ideal scenario, classifiers entry should be removed before breaking changes to prevent such cases. 
The issue is more to keep requirements with tested/working boundaries.
On python3.6 some older version of requests still satisfies requirements - if it still works, then 3.6 should be fine. Otherwise, specifying 3.7 as minimal version should be way to go.

My python3.5 problem had simple workaround to downgrade version manually - currently it's ok for me.

**Comment by mahtin** [comment]
This has been fixed as of `2.19.4`. Sorry about causing this issue.
